src/Types.re

```diff
@@ -78,14 +78,44 @@ type operation = {
 };
 
 /* The result of the GraphQL operation. */
-type operationResult('data) = {
+type operationResultJs('jsData) = {
   operation,
-  data: Js.Nullable.t('data),
+  data: Js.Nullable.t('jsData),
   error: option(CombinedError.combinedErrorJs),
   extensions: option(Js.Dict.t(string)),
   stale: option(bool),
 };
 
+/* The result of the GraphQL operation. */
+type operationResult('data) = {
+  data: option('data),
+  error: option(CombinedError.t),
+  extensions: option(Js.Dict.t(string)),
+  response: operationResponse('data),
+  stale: option(bool),
+}
+and operationResponse('data) =
```

Comment:
Is there a benefit to the mutually recursive type here created by the use of `and`? It seems adequate to me to just specify `operationResponse` as its own standalone type and reference it in `response` on L94. I'd only advocate using a mutually recursive type if there was mutual recursion between these types — as far as I can tell, there isn't currently 🤔

Reply:
You're right.